File: lib/models/trainer.dart
/// نموذج بيانات المدرب - Trainer Model
class Trainer {
  final String id;
  final String name;
  final String specialty;
  final String image;
  final double rating;
  final int experience; // بالسنوات
  final int clients; // عدد العملاء
  final String description;
  final List<String> certifications;
  final bool isAvailable;
  final String? email;
  final String? phone;

  const Trainer({
    required this.id,
    required this.name,
    required this.specialty,
    required this.image,
    required this.rating,
    required this.experience,
    required this.clients,
    required this.description,
    this.certifications = const [],
    this.isAvailable = true,
    this.email,
    this.phone,
  });

  factory Trainer.fromJson(Map<String, dynamic> json) {
    return Trainer(
      id: json['_id']?.toString() ?? json['id']?.toString() ?? '',
      name: json['name'] as String? ?? '',
      specialty: json['specialty'] as String? ?? json['specialization'] as String? ?? '',
      image: json['image'] as String? ?? json['avatar'] as String? ?? '',
      rating: (json['rating'] as num?)?.toDouble() ?? 0.0,
      experience: json['experience'] as int? ?? 0,
      clients: json['clients'] as int? ?? json['clientsCount'] as int? ?? 0,
      description: json['description'] as String? ?? json['bio'] as String? ?? '',
      certifications: json['certifications'] != null
          ? List<String>.from(json['certifications'] as List)
          : const [],
      isAvailable: json['isAvailable'] as bool? ?? json['available'] as bool? ?? true,
      email: json['email'] as String?,
      phone: json['phone'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'specialty': specialty,
      'image': image,
      'rating': rating,
      'experience': experience,
      'clients': clients,
      'description': description,
      'certifications': certifications,
      'isAvailable': isAvailable,
      'email': email,
      'phone': phone,
    };
  }

  /// نسخ مع تعديل
  Trainer copyWith({
    String? id,
    String? name,
    String? specialty,
    String? image,
    double? rating,
    int? experience,
    int? clients,
    String? description,
    List<String>? certifications,
    bool? isAvailable,
    String? email,
    String? phone,
  }) {
    return Trainer(
      id: id ?? this.id,
      name: name ?? this.name,
      specialty: specialty ?? this.specialty,
      image: image ?? this.image,
      rating: rating ?? this.rating,
      experience: experience ?? this.experience,
      clients: clients ?? this.clients,
      description: description ?? this.description,
      certifications: certifications ?? this.certifications,
      isAvailable: isAvailable ?? this.isAvailable,
      email: email ?? this.email,
      phone: phone ?? this.phone,
    );
  }
}

File: backend/models/Trainer.js
const mongoose = require('mongoose');

const trainerSchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Please add a trainer name'],
    trim: true,
  },
  email: {
    type: String,
    required: [true, 'Please add an email'],
    unique: true,
    match: [
      /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/,
      'Please add a valid email',
    ],
  },
  phone: {
    type: String,
    required: [true, 'Please add a phone number'],
  },
  specialization: {
    type: [String],
    required: [true, 'Please add specializations'],
    enum: ['قوة', 'كارديو', 'يوغا', 'تغذية', 'تأهيل', 'كروسفيت', 'بناء أجسام'],
  },
  bio: {
    type: String,
    required: [true, 'Please add a bio'],
  },
  experience: {
    type: Number,
    required: [true, 'Please add years of experience'],
  },
  certifications: [String],
  image: {
    type: String,
    default: 'default-trainer.jpg',
  },
  rating: {
    type: Number,
    default: 0,
    min: 0,
    max: 5,
  },
  availability: {
    type: Map,
    of: [String], // day: ['09:00-12:00', '14:00-18:00']
    default: {},
  },
  clients: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
  }],
  isActive: {
    type: Boolean,
    default: true,
  },
}, {
  timestamps: true,
});

module.exports = mongoose.model('Trainer', trainerSchema);
